TelCo.ColorCoder/Testing.cs

```diff
@@ -1,48 +1,49 @@
+                    Debug.Assert(testPair.minorColor == PairsDataModel.colorMapMinor[j]);
+
+                    testPair = new PairsDataModel.ColorPair()
+                    {
+                        majorColor = PairsDataModel.colorMapMajor[i],
+                        minorColor = PairsDataModel.colorMapMinor[j]
+                    };
+                    int pairNumberReceived = GetPair.GetPairNumberFromColor(testPair);
+                    Debug.Assert(pair_no == pairNumberReceived);
+
+                    pair_no++;
+                }
+            }
+        }
+
         private static void Main(string[] args)
         {
-            int pairNumber = 4;
-            PairsDataModel.ColorPair testPair1 = GetColor.GetColorFromPairNumber(pairNumber);
-            Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
-            Debug.Assert(testPair1.majorColor == Color.White);
-            Debug.Assert(testPair1.minorColor == Color.Brown);
-
-            pairNumber = 5;
-            testPair1 = GetColor.GetColorFromPairNumber(pairNumber);
-            Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
-            Debug.Assert(testPair1.majorColor == Color.White);
-            Debug.Assert(testPair1.minorColor == Color.SlateGray);
-
-            pairNumber = 23;
-            testPair1 = GetColor.GetColorFromPairNumber(pairNumber);
-            Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
-            Debug.Assert(testPair1.majorColor == Color.Violet);
-            Debug.Assert(testPair1.minorColor == Color.Green);
-
-            PairsDataModel.ColorPair testPair2 = new PairsDataModel.ColorPair() { majorColor = Color.Yellow, minorColor = Color.Green };
-            pairNumber = GetPair.GetPairNumberFromColor(testPair2);
-            Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}\n", testPair2, pairNumber);
-            Debug.Assert(pairNumber == 18);
-
-            testPair2 = new PairsDataModel.ColorPair() { majorColor = Color.Red, minorColor = Color.Blue };
-            pairNumber = GetPair.GetPairNumberFromColor(testPair2);
-            Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}", testPair2, pairNumber);
-            Debug.Assert(pairNumber == 6);
+            ManualTesting();
+
+            //ManualDelegate _manualPrintingAddress = new ManualDelegate(ReferenceManual.PrintManual);
+            ReferenceManual.PrintManual(new PrintOnConsole());
+            //ManualDelegate _manualTestingAddress = new ManualDelegate(ReferenceManual.ManualTesting);
+            //ReferenceManual.Manual(ReferenceManual.ManualTesting);
+
+            PrintOnConsoleTest testPrintonConsole = new PrintOnConsoleTest();
+            ReferenceManual.PrintManual(testPrintonConsole);
+            Debug.Assert(testPrintonConsole.timesPrintOnConsoleCalled == 25);

```

Comment:
Next time, try to capture what's important for the customer. Is it enough if the manual has 25 entries? The best test is the one that guarantees customer need. Writing such a test helps you to be very clear on all assumptions about the requirement. Try it out.

Reply:
Okay Sir, will work on it.

Reply:
@sudeeprp , did you mean I change 25 to (no. of major colors)*(no of minor colors)*timesPrintManualCalled.